components/batch-system/src/router.rs

```diff
@@ -54,9 +58,15 @@ where
             control_box,
             normal_scheduler,
             control_scheduler,
+            shut: Arc::new(AtomicBool::new(false)),
         }
     }
 
+    /// The `Router` has been already shut or not.
+    pub fn is_shut(&self) -> bool {
+        self.shut.load(Ordering::Acquire)
```

Comment:
One side is `Acquire` and the other side is `SeqCst` looks weird.
`Acquire` and `Release` should be a twain. Using `SeqCst` both is also ok because it's not a hot path.